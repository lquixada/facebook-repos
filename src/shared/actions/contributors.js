export const CONTRIBUTORS_REQUESTED = 'CONTRIBUTORS_REQUESTED';
export const CONTRIBUTORS_SUCCEEDED = 'CONTRIBUTORS_SUCCEEDED';
export const CONTRIBUTORS_FAILED = 'CONTRIBUTORS_FAILED';

export const MORE_CONTRIBUTORS_REQUESTED = 'MORE_CONTRIBUTORS_REQUESTED';
export const MORE_CONTRIBUTORS_SUCCEEDED = 'MORE_CONTRIBUTORS_SUCCEEDED';
export const MORE_CONTRIBUTORS_FAILED = 'MORE_CONTRIBUTORS_FAILED';

export const fetchContributors = (repoName) => ({
  type: CONTRIBUTORS_REQUESTED,
  payload: {
    repoName
  }
});

export const fetchContributorsSucceeded = (repoName, data) => ({
  type: CONTRIBUTORS_SUCCEEDED,
  payload: {
    repoName,
    data
  }
});

export const fetchContributorsFailed = (repoName, error) => ({
  type: CONTRIBUTORS_FAILED,
  payload: {
    repoName,
    error
  }
});

export const fetchMoreContributors = (repoName, nextUrl) => ({
  type: MORE_CONTRIBUTORS_REQUESTED,
  payload: {
    repoName,
    nextUrl
  }
});

export const fetchMoreContributorsSucceeded = (repoName, data) => ({
  type: MORE_CONTRIBUTORS_SUCCEEDED,
  payload: {
    repoName,
    data
  }
});

export const fetchMoreContributorsFailed = (repoName, error) => ({
  type: MORE_CONTRIBUTORS_FAILED,
  payload: {
    repoName,
    error
  }
});
