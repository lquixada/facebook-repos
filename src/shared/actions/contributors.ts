export const CONTRIBUTORS_REQUESTED = 'CONTRIBUTORS_REQUESTED'
export const CONTRIBUTORS_SUCCEEDED = 'CONTRIBUTORS_SUCCEEDED'
export const CONTRIBUTORS_FAILED = 'CONTRIBUTORS_FAILED'

export const fetchContributors = (payload) => ({
  payload,
  type: CONTRIBUTORS_REQUESTED,
})

export const fetchContributorsSucceeded = (payload: {owner: string, repoName: string, data: any}) => ({
  payload,
  type: CONTRIBUTORS_SUCCEEDED,
})

export const fetchContributorsFailed = (payload: {owner: string, repoName: string, error: string}) => ({
  payload,
  type: CONTRIBUTORS_FAILED,
})
