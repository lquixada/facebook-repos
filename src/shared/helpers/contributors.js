import fetch from 'cross-fetch';

import config from '../config';

export const fetchContributors = async (repoName) => {
  const endpoint = config.endpoints.contributors;
  const url = `${endpoint.replace('%{repo}', repoName)}?per_page=40`;
  const res = await fetch(url);
  return await res.json();
};
