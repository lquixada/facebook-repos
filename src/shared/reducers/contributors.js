/* eslint-disable complexity */
import merge from 'lodash.merge';

import {
  CONTRIBUTORS_SUCCEEDED, CONTRIBUTORS_FAILED, CONTRIBUTORS_REQUESTED,
  MORE_CONTRIBUTORS_SUCCEEDED, MORE_CONTRIBUTORS_FAILED, MORE_CONTRIBUTORS_REQUESTED
} from '../actions';

const result = (state = [], action) => {
  const {payload, type} = action;

  switch (type) {
    case CONTRIBUTORS_SUCCEEDED:
      return payload.data;
    case MORE_CONTRIBUTORS_SUCCEEDED:
      return state.result.concat(payload.data);
    default:
      return state;
  }
};

export default function contributors(state = {}, action) {
  const {payload, type, error} = action;

  switch (type) {
    case CONTRIBUTORS_REQUESTED:
    case MORE_CONTRIBUTORS_REQUESTED:
      return merge({}, state, {
        [payload.repoName]: {
          isLoading: true,
          error: null,
        },
      });

    case CONTRIBUTORS_SUCCEEDED:
    case MORE_CONTRIBUTORS_SUCCEEDED:
      return merge({}, state, {
        [payload.repoName]: {
          result: result(state[payload.data], action),
          isLoading: false,
          error: null,
        },
      });

    case CONTRIBUTORS_FAILED:
    case MORE_CONTRIBUTORS_FAILED:
      return merge({}, state, {
        [payload.repoName]: {
          isLoading: false,
          error,
        },
      });

    default:
      return state;
  }
}
