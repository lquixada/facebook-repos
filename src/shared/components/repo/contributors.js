import React from 'react';
import styled from 'styled-components';

import {flex} from '../../helpers';

export class Contributors extends React.Component {
  constructor(props) {
    super(props);
    this.fetchNext = this.fetchNext.bind(this);
  }

  componentDidMount() {
    this.fetch();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.repoName !== this.props.repoName) {
      this.fetch();
    }
  }

  fetch() {
    if (!this.hasContributorsLoaded()) {
      this.props.fetchContributors(this.props.repoName);
    }
  }

  fetchNext() {
    const {repoName, contributors} = this.props;
    this.props.fetchMoreContributors(repoName, contributors.next);
  }

  hasContributorsLoaded() {
    const {contributors} = this.props;
    return contributors && contributors.result && contributors.result.length !== 0;
  }

  hasMoreContributors() {
    return !!this.props.contributors.next;
  }

  isLoading() {
    return this.props.contributors.isLoading;
  }

  getCount() {
    const {count} = this.props;
    return count? count[1] : 'error';
  }

  render() {
    return (
      <Wrapper>
        <Title>Contributors ({this.getCount()})</Title>
        {this.renderContent()}
      </Wrapper>
    );
  }

  renderContent() {
    if (!this.hasContributorsLoaded()) {
      return (
        <div>Loading...</div>
      );
    }

    return (
      <div>
        <List>
          {this.props.contributors.result.map((contributor, i) =>
            <Item key={contributor.login+i}>
              <Link href={contributor.html_url}>
                <Image src={contributor.avatar_url} />
                <Text>{contributor.login}</Text>
              </Link>
            </Item>
          )}
        </List>
        {this.hasMoreContributors() &&
          <Button onClick={this.fetchNext}>
            {this.isLoading()? 'Loading...' : 'More'}
          </Button>
        }
      </div>
    );
  }
}

const Wrapper = styled.div`
  margin: 2rem 0;
`;

const Title = styled.h3`
  color: #767676;
  font-size: 1.4rem;
  font-weight: normal;
  line-height: 1.5;
`;

const List = styled.ul`
  ${flex.display()}
  ${flex.left()}
  ${flex.flow('row', 'wrap')};
  margin-top: 1rem;
`;

const Item = styled.li`
  ${flex.display()}
  ${flex.left()}
  ${flex.middle()}
  margin-bottom: 1rem;
  width: 25%;
  font-size: 1.2rem;
  white-space: nowrap;
`;

const Link = styled.a`
  ${flex.display()}
  ${flex.left()}
  ${flex.middle()}
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;

const Image = styled.img`
  margin-right: 1rem;
  width: 4rem;
  height: 4rem;
  border-radius: 2rem;
  overflow: hidden;
`;

const Text = styled.span`
  display: inline-block;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;

const Button = styled.button`
  display: block;
  width: 100%;
  padding: .6rem;
  margin-top: 2rem;
  font-size: 1.4rem;
  font-weight: 600;
  color: #0366d6;
  background: #fff;
  border: 1px solid #e1e4e8;
  border-radius: .3rem;
  cursor: pointer;

  &:hover, &:focus {
    background-color: #f6f8fa;
  }
`;
