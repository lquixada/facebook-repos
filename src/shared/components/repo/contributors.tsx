import React from 'react'
import {NavLink} from 'react-router-dom'
import styled from 'styled-components'

import {flex, tablets} from '../../helpers'

interface IContributor {
  login: string
  avatar_url: string
  html_url: string
}
interface IProps {
  data: IContributor[]
  total: string | number
  hasLoaded: boolean
  hasMore: boolean
  isLoadingMore: boolean
  onNext(f: any): any
}

export const Contributors = (props: IProps) => (
  <Wrapper>
    <Title>Contributors ({props.total})</Title>
    {!props.hasLoaded ? <Loading /> : <Content {...props} />}
  </Wrapper>
)

const Loading = () => (
  <div>Loading...</div>
)

const Content = ({data, hasMore, isLoadingMore, onNext}: IProps) => (
  <div>
    <List>
      {data.map((contributor, i) =>
        <Item key={contributor.login + i}>
          <Link to={`/r/${contributor.login}`}>
            <Image alt={`User ${contributor.login}`} src={`${contributor.avatar_url}&s=40`} />
            <Text>{contributor.login}</Text>
          </Link>
        </Item>,
      )}
    </List>
    {hasMore &&
      <Button onClick={onNext}>
        {isLoadingMore ? 'Loading...' : 'More'}
      </Button>
    }
  </div>
)

const Wrapper = styled.div`
  margin: 2rem 0;
`

const Title = styled.h3`
  color: #767676;
  font-size: 1.4rem;
  font-weight: normal;
  line-height: 1.5;
`

const List = styled.ul`
  ${flex.display()}
  ${flex.left()}
  ${flex.flow('row', 'wrap')};
  margin-top: 1rem;
`

const Item = styled.li`
  ${flex.display()}
  ${flex.left()}
  ${flex.middle()}
  margin-bottom: 1rem;
  width: 25%;
  font-size: 1.2rem;
  white-space: nowrap;

  @media (${tablets}) {
    width: 33.33%;
  }
`

const Link = styled(NavLink)`
  ${flex.display()}
  ${flex.left()}
  ${flex.middle()}
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`

const Image = styled.img`
  margin-right: 1rem;
  width: 4rem;
  height: 4rem;
  border-radius: 2rem;
  overflow: hidden;
`

const Text = styled.span`
  display: inline-block;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`

const Button = styled.button`
  display: block;
  width: 100%;
  padding: .6rem;
  margin-top: 2rem;
  font-size: 1.4rem;
  font-weight: 600;
  color: #0366d6;
  background: #fff;
  border: 1px solid #e1e4e8;
  border-radius: .3rem;
  cursor: pointer;

  &:hover, &:focus {
    background-color: #f6f8fa;
  }
`
