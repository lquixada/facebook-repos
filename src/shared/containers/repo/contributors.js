import React from 'react'
import {connect} from 'react-redux'
import {List} from 'immutable'

import {Contributors} from '../../components/repo/contributors'
import * as actions from '../../actions'

export class ContributorsContainer extends React.Component {
  constructor (props) {
    super(props)
    this.fetchNext = this.fetchNext.bind(this)
  }

  fetchNext () {
    this.props.fetchContributors({
      owner: this.props.owner,
      repoName: this.props.repoName
    })
  }

  hasLoaded () {
    const {contributors} = this.props
    return contributors && contributors.get('data') && !contributors.get('data').isEmpty()
  }

  hasMore () {
    return !!this.props.contributors.get('nextPage')
  }

  isLoading () {
    return this.props.contributors.get('isLoading')
  }

  getCount () {
    const {count} = this.props
    return count ? count.get(1) : 'error'
  }

  render () {
    if (!this.hasLoaded()) {
      return null
    }

    return (
      <Contributors
        total={this.getCount()}
        data={this.props.contributors.get('data')}
        isLoadingMore={this.isLoading()}
        hasLoaded={this.hasLoaded()}
        hasMore={this.hasMore()}
        onNext={this.fetchNext}
      />
    )
  }
}

const mapStateToProps = ({contributors, counts}, {owner, repoName}) => ({
  // REMEMBER: "counts" has the following
  // scheme [[repoName1, count1, false], [repoName2, count2, false]]
  count: counts.get('data', List()).find((count) => count.get(0) === repoName),
  contributors: contributors.getIn([owner, repoName])
})

export default connect(mapStateToProps, actions)(ContributorsContainer)
