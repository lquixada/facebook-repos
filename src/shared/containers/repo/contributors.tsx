import {List, Map} from 'immutable'
import React from 'react'
import {connect} from 'react-redux'

import * as actions from '../../actions'
import {Contributors} from '../../components/repo/contributors'

interface IProps {
  owner: string
  repoName: string
  contributors: Map<string, any>
  count: List<string>
  fetchContributors(payload: object): void
  onNext(f: any): any
}

export class ContributorsContainer extends React.Component<IProps, any> {
  constructor(props) {
    super(props)
    this.fetchNext = this.fetchNext.bind(this)
  }

  public fetchNext() {
    this.props.fetchContributors({
      owner: this.props.owner,
      repoName: this.props.repoName,
    })
  }

  public hasLoaded() {
    const {contributors} = this.props
    return contributors && contributors.get('data') && !contributors.get('data').isEmpty()
  }

  public hasMore() {
    return !!this.props.contributors.get('nextPage')
  }

  public isLoading(): boolean {
    return this.props.contributors.get('isLoading')
  }

  public getCount(): string {
    const {count} = this.props
    return count ? count.get(1) : 'error'
  }

  public render() {
    if (!this.hasLoaded()) {
      return null
    }

    return (
      <Contributors
        total={this.getCount()}
        data={this.props.contributors.get('data')}
        isLoadingMore={this.isLoading()}
        hasLoaded={this.hasLoaded()}
        hasMore={this.hasMore()}
        onNext={this.fetchNext}
      />
    )
  }
}

const mapStateToProps = ({contributors, counts}, {owner, repoName}) => ({
  contributors: contributors.getIn([owner, repoName]),
  // REMEMBER: "counts" has the following
  // scheme [[repoName1, count1, false], [repoName2, count2, false]]
  count: counts.get('data', List()).find((count) => count.get(0) === repoName),
})

export default connect(mapStateToProps, actions)(ContributorsContainer)
