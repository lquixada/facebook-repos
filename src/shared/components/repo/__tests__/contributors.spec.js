import React from 'react';
import {mount} from 'enzyme';

import {Contributors} from '../contributors';

describe('<Contributors />', () => {
  it('list contributors', () => {
    const data = [
      {login: 'lquixada1', html_url: 'https://lquixada1/",', avatar: 'https://avatars1/'},
      {login: 'lquixada2', html_url: 'https://lquixada2/",', avatar: 'https://avatars2/'},
      {login: 'lquixada3', html_url: 'https://lquixada3/",', avatar: 'https://avatars3/'}
    ];
    const component = mount(<Contributors
      total={3}
      data={data}
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={(f) => f}
    />);
    expect(component).toMatchSnapshot();
  });

  it('shows loading', () => {
    const component = mount(<Contributors total={3} hasLoaded={false} />);
    expect(component).toMatchSnapshot();
  });

  it('show button More', () => {
    const component = mount(<Contributors
      total={3}
      data={[]}
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={(f) => f}
    />);
    expect(component).toMatchSnapshot();
  });

  it('show button Loading', () => {
    const component = mount(<Contributors
      total={3}
      data={[]}
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={(f) => f}
    />);
    expect(component).toMatchSnapshot();
  });
});
