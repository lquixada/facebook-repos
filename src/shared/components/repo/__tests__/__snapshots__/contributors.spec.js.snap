// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contributors /> list contributors 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Contributors
      data={
        Immutable.List [
          Immutable.Map {
            "login": "user1",
            "html_url": "https://user1/\\",",
            "avatar": "https://avatars1/",
          },
          Immutable.Map {
            "login": "user2",
            "html_url": "https://user2/\\",",
            "avatar": "https://avatars2/",
          },
          Immutable.Map {
            "login": "user3",
            "html_url": "https://user3/\\",",
            "avatar": "https://avatars3/",
          },
        ]
      }
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={[Function]}
      total={3}
    >
      <styled.div>
        <div
          className="sc-bdVaJa LrVZr"
        >
          <styled.h3>
            <h3
              className="sc-bwzfXH iAozsW"
            >
              Contributors (
              3
              )
            </h3>
          </styled.h3>
          <Content
            data={
              Immutable.List [
                Immutable.Map {
                  "login": "user1",
                  "html_url": "https://user1/\\",",
                  "avatar": "https://avatars1/",
                },
                Immutable.Map {
                  "login": "user2",
                  "html_url": "https://user2/\\",",
                  "avatar": "https://avatars2/",
                },
                Immutable.Map {
                  "login": "user3",
                  "html_url": "https://user3/\\",",
                  "avatar": "https://avatars3/",
                },
              ]
            }
            hasLoaded={true}
            hasMore={true}
            isLoadingMore={false}
            onNext={[Function]}
            total={3}
          >
            <div>
              <styled.ul>
                <ul
                  className="sc-htpNat hppgGZ"
                >
                  <styled.li
                    key="user10"
                  >
                    <li
                      className="sc-bxivhb gUOEDj"
                    >
                      <Styled(NavLink)
                        to="/r/user1"
                      >
                        <NavLink
                          activeClassName="active"
                          aria-current="page"
                          className="sc-ifAKCX ibOcjp"
                          to="/r/user1"
                        >
                          <Route
                            path="\\\\/r\\\\/user1"
                          >
                            <Link
                              aria-current={null}
                              className="sc-ifAKCX ibOcjp"
                              replace={false}
                              to="/r/user1"
                            >
                              <a
                                aria-current={null}
                                className="sc-ifAKCX ibOcjp"
                                href="/r/user1"
                                onClick={[Function]}
                              >
                                <styled.img
                                  src="undefined&s=40"
                                >
                                  <img
                                    className="sc-EHOje dcMEop"
                                    src="undefined&s=40"
                                  />
                                </styled.img>
                                <styled.span>
                                  <span
                                    className="sc-bZQynM fGQOqI"
                                  >
                                    user1
                                  </span>
                                </styled.span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </Styled(NavLink)>
                    </li>
                  </styled.li>
                  <styled.li
                    key="user21"
                  >
                    <li
                      className="sc-bxivhb gUOEDj"
                    >
                      <Styled(NavLink)
                        to="/r/user2"
                      >
                        <NavLink
                          activeClassName="active"
                          aria-current="page"
                          className="sc-ifAKCX ibOcjp"
                          to="/r/user2"
                        >
                          <Route
                            path="\\\\/r\\\\/user2"
                          >
                            <Link
                              aria-current={null}
                              className="sc-ifAKCX ibOcjp"
                              replace={false}
                              to="/r/user2"
                            >
                              <a
                                aria-current={null}
                                className="sc-ifAKCX ibOcjp"
                                href="/r/user2"
                                onClick={[Function]}
                              >
                                <styled.img
                                  src="undefined&s=40"
                                >
                                  <img
                                    className="sc-EHOje dcMEop"
                                    src="undefined&s=40"
                                  />
                                </styled.img>
                                <styled.span>
                                  <span
                                    className="sc-bZQynM fGQOqI"
                                  >
                                    user2
                                  </span>
                                </styled.span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </Styled(NavLink)>
                    </li>
                  </styled.li>
                  <styled.li
                    key="user32"
                  >
                    <li
                      className="sc-bxivhb gUOEDj"
                    >
                      <Styled(NavLink)
                        to="/r/user3"
                      >
                        <NavLink
                          activeClassName="active"
                          aria-current="page"
                          className="sc-ifAKCX ibOcjp"
                          to="/r/user3"
                        >
                          <Route
                            path="\\\\/r\\\\/user3"
                          >
                            <Link
                              aria-current={null}
                              className="sc-ifAKCX ibOcjp"
                              replace={false}
                              to="/r/user3"
                            >
                              <a
                                aria-current={null}
                                className="sc-ifAKCX ibOcjp"
                                href="/r/user3"
                                onClick={[Function]}
                              >
                                <styled.img
                                  src="undefined&s=40"
                                >
                                  <img
                                    className="sc-EHOje dcMEop"
                                    src="undefined&s=40"
                                  />
                                </styled.img>
                                <styled.span>
                                  <span
                                    className="sc-bZQynM fGQOqI"
                                  >
                                    user3
                                  </span>
                                </styled.span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </Styled(NavLink)>
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
              <styled.button
                onClick={[Function]}
              >
                <button
                  className="sc-gzVnrw kKiVFV"
                  onClick={[Function]}
                >
                  More
                </button>
              </styled.button>
            </div>
          </Content>
        </div>
      </styled.div>
    </Contributors>
  </Router>
</MemoryRouter>
`;

exports[`<Contributors /> show button Loading 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Contributors
      data={Array []}
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={[Function]}
      total={3}
    >
      <styled.div>
        <div
          className="sc-bdVaJa LrVZr"
        >
          <styled.h3>
            <h3
              className="sc-bwzfXH iAozsW"
            >
              Contributors (
              3
              )
            </h3>
          </styled.h3>
          <Content
            data={Array []}
            hasLoaded={true}
            hasMore={true}
            isLoadingMore={false}
            onNext={[Function]}
            total={3}
          >
            <div>
              <styled.ul>
                <ul
                  className="sc-htpNat hppgGZ"
                />
              </styled.ul>
              <styled.button
                onClick={[Function]}
              >
                <button
                  className="sc-gzVnrw kKiVFV"
                  onClick={[Function]}
                >
                  More
                </button>
              </styled.button>
            </div>
          </Content>
        </div>
      </styled.div>
    </Contributors>
  </Router>
</MemoryRouter>
`;

exports[`<Contributors /> show button More 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Contributors
      data={Array []}
      hasLoaded={true}
      hasMore={true}
      isLoadingMore={false}
      onNext={[Function]}
      total={3}
    >
      <styled.div>
        <div
          className="sc-bdVaJa LrVZr"
        >
          <styled.h3>
            <h3
              className="sc-bwzfXH iAozsW"
            >
              Contributors (
              3
              )
            </h3>
          </styled.h3>
          <Content
            data={Array []}
            hasLoaded={true}
            hasMore={true}
            isLoadingMore={false}
            onNext={[Function]}
            total={3}
          >
            <div>
              <styled.ul>
                <ul
                  className="sc-htpNat hppgGZ"
                />
              </styled.ul>
              <styled.button
                onClick={[Function]}
              >
                <button
                  className="sc-gzVnrw kKiVFV"
                  onClick={[Function]}
                >
                  More
                </button>
              </styled.button>
            </div>
          </Content>
        </div>
      </styled.div>
    </Contributors>
  </Router>
</MemoryRouter>
`;

exports[`<Contributors /> shows loading 1`] = `
<Contributors
  hasLoaded={false}
  total={3}
>
  <styled.div>
    <div
      className="sc-bdVaJa LrVZr"
    >
      <styled.h3>
        <h3
          className="sc-bwzfXH iAozsW"
        >
          Contributors (
          3
          )
        </h3>
      </styled.h3>
      <Loading>
        <div>
          Loading...
        </div>
      </Loading>
    </div>
  </styled.div>
</Contributors>
`;
